chess/board_impl.h

```diff
@@ -51,6 +54,9 @@ namespace space {
 		bool m_canBlackCastleRight;
 		Color m_whoPlaysNext;
 		bool checkObstructions(Move m) const;
+		bool canMove(Move m) const;
+		bool checkPathEmpty(Move m) const;
```

Comment:
Just for my curiosity: Why did you add these functions? They are not used anywhere, are they?

Reply:
Added these two for alternate implementation of isUnderCheck, also can be useful for identifying attacks & defenses (not implemented yet)